src/sqlfluff/core/config.py

```diff
@@ -429,15 +438,32 @@ def __init__(
         # Dialect and Template selection.
         # NB: We import here to avoid a circular references.
         from sqlfluff.core.dialects import dialect_selector
-        from sqlfluff.core.templaters import templater_selector
 
         self._configs["core"]["dialect_obj"] = dialect_selector(
             self._configs["core"]["dialect"]
         )
-        self._configs["core"]["templater_obj"] = templater_selector(
+        self._configs["core"]["templater_obj"] = self.get_templater(
             self._configs["core"]["templater"]
         )
 
+    def __getstate__(self):
+        # Copy the object's state from self.__dict__ which contains
+        # all our instance attributes. Always use the dict.copy()
+        # method to avoid modifying the original state.
+        state = self.__dict__.copy()
+        # Remove the unpicklable entries.
+        del state["_plugin_manager"]
```

Comment:
IIUC, this would prevent using "not installed" linter rules in combination with `--processes`. See #1233. I'm not sure we need to solve this, but maybe need to document it when #1233 is implemented.

Reply:
Random thought: I wonder if there may be a way to convert the plugin manager instance to a pickleable format, e.g. for each plugin, get its module and class/function name, then on the other side, create an empty plug-in manager and build it up again. It wouldn't be guaranteed to work in every case, but my intuition is that it could handle 99% of cases. Worth creating an issue for this? (Or adding these notes to #1233?)

Reply:
Agreed. I actually wonder whether it might be more practical to pickle any use defined plugins and pass them to the processes rather than passing around the plugin manager? It would need to be implemented with #1233 but might be more practical. For now I'll add in a warning in the code.